fe/src/main/java/org/apache/doris/analysis/DataDescription.java

```diff
@@ -189,92 +186,100 @@ public boolean isPullLoad() {
-        for (Expr expr : columnMappingList) {
-            if (!(expr instanceof BinaryPredicate)) {
-                throw new AnalysisException("Mapping function expr error. expr: " + expr.toSql());
-            }
+        for (Expr columnExpr : columnMappingList) {
 
-            BinaryPredicate predicate = (BinaryPredicate) expr;
+            if (!(columnExpr instanceof BinaryPredicate)) {
+                throw new AnalysisException("Mapping function expr only support the column or eq binary predicate. "
+                                                    + "Expr: " + columnExpr.toSql());
+            }
+            BinaryPredicate predicate = (BinaryPredicate) columnExpr;
             if (predicate.getOp() != Operator.EQ) {
-                throw new AnalysisException("Mapping function operator error. op: " + predicate.getOp());
+                throw new AnalysisException("Mapping function expr only support the column or eq binary predicate. "
+                                                    + "The mapping operator error, op: " + predicate.getOp());
             }
-
             Expr child0 = predicate.getChild(0);
             if (!(child0 instanceof SlotRef)) {
-                throw new AnalysisException("Mapping column error. column: " + child0.toSql());
+                throw new AnalysisException("Mapping function expr only support the column or eq binary predicate. "
+                                                    + "The mapping column error. column: " + child0.toSql());
             }
-
             String column = ((SlotRef) child0).getColumnName();
-            if (columnToFunction.containsKey(column)) {
+            if (!columnNames.add(column)) {
                 throw new AnalysisException("Duplicate column mapping: " + column);
             }
-
-            // we support function and column reference to change a column name
+            // hadoop load only supports the FunctionCallExpr
             Expr child1 = predicate.getChild(1);
-            if (!(child1 instanceof FunctionCallExpr)) {
-                if (isPullLoad && child1 instanceof SlotRef) {
-                    // we only support SlotRef in pull load
-                } else {
-                    throw new AnalysisException("Mapping function error, function: " + child1.toSql());
-                }
-            }
-
-            if (!child1.supportSerializable()) {
-                throw new AnalysisException("Expr do not support serializable." + child1.toSql());
+            if (!isPullLoad && !(child1 instanceof FunctionCallExpr)) {
+                throw new AnalysisException("Hadoop load only supports the designated function. "
+                                                    + "The error mapping function is:" + child1.toSql());
             }
+            ImportColumnDesc importColumnDesc = new ImportColumnDesc(column, child1);
+            parsedColumnExprList.add(importColumnDesc);
+            analyzeColumnToFunction(column, child1);
```

Comment:
If this is not a hadoop load, do we need to call this function ?

Reply:
Actually, the columnToFunction is also used by the check function.
The check function is mainly used for checking the correctness of column expr.
Example: the column which is not defined in columns and column mapping has default value or not.
So maybe the columnToFunction still has been used in hadoop and broker load.
The annotation of columnToFunction need to be changed.